src/ert/_c_wrappers/enkf/analysis_config.py

```diff
@@ -61,42 +42,46 @@ def __init__(
         self._analysis_iter_config = analysis_iter_config or AnalysisIterConfig()
         self._update_log_path = update_log_path
 
-        self._analysis_select = analysis_select
-        self._analysis_set_var = analysis_set_var or []
-        self._analysis_copy = analysis_copy or []
+        analysis_set_var = analysis_set_var or []
+        analysis_copy = analysis_copy or []
+        es_module = AnalysisModule.ens_smoother_module()
+        ies_module = AnalysisModule.iterated_ens_smoother_module()
+        self._modules: Dict[str, AnalysisModule] = {
+            AnalysisMode.ENSEMBLE_SMOOTHER: es_module,
+            AnalysisMode.ITERATED_ENSEMBLE_SMOOTHER: ies_module,
+        }
+        self._active_module = analysis_select or AnalysisMode.ENSEMBLE_SMOOTHER
+        # copy modules
+        for element in analysis_copy:
```

Comment:
have you considered instead of creating an comment explaining the functionality to extract a private func `_copy_modules()` ? Not saying it is better, but sometimes this could be beneficial keeping the init more clean...